internal/provider/resource_gitlab_tag.go

```diff
@@ -0,0 +1,123 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_tag", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage GitLab tags.",
+
+		CreateContext: resourceGitlabTagCreate,
+		ReadContext:   resourceGitlabTagRead,
+		DeleteContext: resourceGitlabTagDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
```

Comment:
There are missing API fields here, e.g. `release`, `protected`, ...

There is also `target`, which seems to be the `ref`. The upstream API is inconsistent here. I would opt for keeping the attribute here `ref`, but documenting that it corresponds to the `target` field in the upstream API.

Reply:
I have bad news for you. [Go-Gitlab](https://github.com/xanzy/go-gitlab) return only:
```
{
  Commit: {
    ID:"07b712c4883bd867c80582b7a8d15495cccaaa2e",
    ShortID:"07b712c4",
    Title:"Initialized from 'Ruby on Rails' project template",
    CreatedAt: {},
    ParentIDs: [],
    Message: "Initialized from 'Ruby on Rails' project template",
    AuthorName:"GitLab",
    AuthorEmail:"root@localhost",
    AuthoredDate: {},
    CommitterName: "Hordur Freyr Yngvason",
    CommitterEmail: "hfyngvason@gitlab.com",
    CommittedDate: {},
    ProjectID: 0,
    WebURL: "http://12ac70e67436/terraform/myrails/-/commit/07b712c4883bd867c80582b7a8d15495cccaaa2e"
  },
  Name: "test3",
  Message: ""
}
```
I know that [Gitlab API](https://docs.gitlab.com/ee/api/tags.html) provides you with other fields but I can't add them without contributing to [Go-Gitlab](https://github.com/xanzy/go-gitlab).

Reply:
Feel free to contribute to go-gitlab to implement these. Otherwise, I'm up for merging this prior and add an issue to align the APIs ;)